### PR TITLE
TypeScript compilation in NativeScript

### PR DESCRIPTION
**Pull request by teobugslayer**
see https://github.com/NativeScript/nativescript-cli/issues/421

### CODE REVIEW

**Comment by ligaz** [comment]
We have been discussing that we might want to add the TypeScript as project level hook and not global one. The problem with the global one is that it cannot easily be overwritten so that users can create their own workflows using task runners like grunt, gulp or broccoli. 

Another benefit of having it per project is that users might use different version of the TypeScript compiler in each project along with different `tsconfig.json`

File: lib/fiber-bootstrap.ts [hunk 1]
```diff
@@ -0,0 +1,15 @@
+import Fiber = require("fibers");
```
Comment (Fatme) [inline lib/fiber-bootstrap.ts]: This file is duplicated and should be moved to common lib

File: lib/hooks/before-livesync.ts [hunk 2]
```diff
@@ -0,0 +1,3 @@
+require("./typescript-compilation.js");
```
Comment (Fatme) [inline lib/hooks/before-livesync.ts]: This is unneeded here

File: lib/hooks/typescript-compilation.ts [hunk 3]
```diff
@@ -0,0 +1,22 @@
+///<reference path="../.d.ts"/>
```
Comment (Fatme) [inline lib/hooks/typescript-compilation.ts]: This service should be in common lib

**Comment by teobugslayer** [comment]
We will support TypeScript per-project, closing this PR for now.

**Pull request closed without merging**